src/Services/SignatureFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace LearnositySdk\Services;
+
+use LearnositySdk\Services\Signatures\SignatureInterface;
+use LearnositySdk\Services\Signatures\HashSignature;
+use LearnositySdk\Services\Signatures\HmacSignature;
+
+class SignatureFactory
+{
+    const LEGACY_SIGNATURE_VERSION = '01';
+
+    const DEFAULT_SIGNATURE_VERSION = '02';
+    /**
+     * @param string $version
+     * @return SignatureInterface
+     * @throws \Exception
+     */
+    public static function getSignatureGenerator(string $version): SignatureInterface
+    {
+        switch ($version) {
+            case self::LEGACY_SIGNATURE_VERSION:
+                $signatureInstance = new HashSignature();
+                break;
+            default:
+                $signatureInstance = new HmacSignature();
+        }
+
+        if (!$signatureInstance instanceof SignatureInterface) {
+            throw new \Exception('No can do, must implement signature interface');
+        }
+
+        return $signatureInstance;
+    }
+
+    /**
+     * @param string $signature
+     * @return string
+     */
+    public static function getSignatureVersion(string $signature): string
+    {
+        $signatureParts = str_split($signature);
```

Comment:
Not needed.

Reply:
Apologies that was left over from an older signature check implementation.